### PR TITLE
Deathsquad tank madafakkassssss

### PR DESCRIPTION
**Pull request by Kmc2000**
Refer to issue https://forums.yogstation.net/index.php?threads/tanks-death-squad-tanks.14645/#post-131734

### Intent of your Pull Request

Oh my God driving this around is so fun

https://gyazo.com/f888537f63baa8f2e68de15849a0df43

Smash

Why go around walls or open airlocks when you can go through them?

#### Changelog

:cl: Kmc2000
rscadd: Added in a tank for the deathsquad, admins, there is a balanced version and a normal, OP one. The OP will destroy everything and the balanced one is killable.
/:cl:

### CODE REVIEW

**Comment by Kmc2000** [comment]
Ree why is it failing I haven't touched the maps

**Comment by Partheo** [comment]
Why does it have a strafe button?
Can it be destroyed? Does it have health?
Can it fire at all?
What happens when it runs over someone?
How do you charge it?

**Comment by Kmc2000** [comment]
It is a mech 
It can be killed, it has 2000 health so glhf
It fires death
It has the same effect as a crusher
You charge it with the relay, or in a mech charger

**Comment by Super3222** [comment]
you could've ported the tanks from NML

edit: but these tanks look cool too

**Comment by ghost** [comment]
who the fuck takes ideas from aphielansin?

why would we add this?

>2000 health

no, that's incredibly stupid. even deathsquad is intended to be beatable, just extremely difficult to do so. at that point, you might as well just del-all mobs and call it a night if you intend to make the DS unbeatable.

**Comment by ShadowDeath6** [comment]
don't pulse rifles do 80 damage

**Comment by Jc1469** [comment]
ftr 2000 health is nearly as much as megafauna.  megafauna dont move that fast and trample everything in their path with player-targetted ranged attacks.

**Comment by Chuggachar** [comment]
The turret is too far back on the hull.

File: code/game/mecha/combat/tank.dm [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+			var/mob/living/M = AM
+			M.say("AHHHH!!!!!!")
+			M.Paralyse(5)
+			M.adjustBruteLoss(smash_damage)
+
+
+/obj/mecha/combat/tank/Move()
+	if(prob(1))
+		var/tempsound = pick(creaks)
+		playsound_global(tempsound,repeat=0, channel=1, volume=100)
```
Comment (Cruix) [inline code/game/mecha/combat/tank.dm:130]: Why is this global?

Comment (Kmc2000) [reply code/game/mecha/combat/tank.dm:130]: Because the floors buckle under its weight, it should give an indication of "Oh fuck what is going on"

**Comment by ghost** [comment]
1. dont use fucking mechs for a tank. you could've just asked me to put this in the car PR
2. what the fuck 2k health
3. s t r a f i n g

**Comment by Kmc2000** [comment]
Asked floyd to put this in his PR, will close when that's in.

**Comment by ghost** [comment]
Maybe destroy two tiles instead of one? Also, it's way too fast. Not that I care about speed, but others won't like it. Maybe make of slower. I get the idea that DS is supposed clean the round up, , but it should still be beatable somehow. Maybe when going through reinforced objects and maybe normal walls, an action bar should show up. "The tank starts plowing into the wall, destroying it." A cool message could be added to whom ever is standing on the other side. "The wall next to you begins to shake; maybe you should step away."

When it comes to abilities, the main attacks should be a cannon and an LMG. There could be counter measures such as smoke, teargas, and a flash. Anyone who is ran over should probably be gibbed, with the driver hearing "You hear a scream, the cracking of bones, and then silence."

I feel the tanks field of view should not be total. Perhaps the view of the driver should be the same as a welding mask. You could open a match, which gives total view. This opens ideas for attacks by the crew, whether by shooting the driver, or tossing a grenade in. I think the grenade part would be cool. EMPs or flashbangs could be tossed in, which would disorient the diver or immobilise the tank, allowing C4 or IEDs to be placed to be placed. This would balance the tank to the point of it being still being a death machine on acid, but still reasonable.

I also have some questions. What will the tank be called? You should have a poll, based off of nominated names from this thread. (My name ideas: Buckthorn II, Corporate IV, Mjiksen Mk3 , or the Rolling Fortress). Will it be space worthy? How will the
Death squad safely transport it from Centcomm? Either it's too big or someone might move it and put a whole in the shuttle, spacing the tank and other squad members. Perhaps make a "Blue space Tank Pill" similar to that of the blue space shelter pill. "A small, red pill with instructions to deploy it with personnel at a safe distance. EDIT: perhaps the pill should be used in the drivers hand, immediately placing them in the tanks seat.

Making multiple types of the tanks seems like a good idea, with just some color changes. Different tanks could have shotguns, repair drones, flamethrowera, and grenade launchers. The tanks could have similar abilities to mechs, such as the Gyaxs ability to speed up, the Phazon's phase ability, and the Durand's lockdown ability.

With all this work, maybe the crew should be able to make these (station goal?). An older, weaker version could be found on Laval and or a space ruin. To introduce the tank, maybe make a new ghost role, called "Syndicate Test Scientists", basically where you can test out the tank and perhaps other syndicate tech. I will make a suggestion thread on this in the near future. The description for could be " A massive, bulky Death Squad tank. While mechs have largely replaced tanks, they're still a few still in use, and are very deadly. Your best chances of survival are to start running."

I think the tank idea is very promising, and should be more than a OP tank that has only a fancy sprite and cannon.

**Comment by Kmc2000** [comment]
There is a balanced version.

There are messages when it plows through a wall.

**Comment by Partheo** [comment]
@onepunchronan has great ideas, I think it's important that you add a bit of polish to this PR rather than just quickly making a tank for the sake of tanks. Specifically, slowing it down and making it take time to break walls. This means it would actually to take hallways.

Agreed with @qustinnus: why a mech? Strafing is retarded. This thing shouldn't be able to strafe.

**Comment by Notamaniac** [comment]
Just play deja vu while strafing and pretend you're drifiting

**Comment by Amelia0010** [comment]
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"
NT-05 tank drives over Frank Rifler and flattens them!
(F) DEAD: Frank Rifler yells, "AHHHH!!!!!!"


You can't run buckled people down

**Comment by Jc1469** [comment]
even if you do run over someone, their ghost screams even when dead

File: code/game/mecha/combat/tank.dm [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		else
+			var/obj/machinery/Q = M
+			Q.take_damage(smash_damage, BRUTE, 0)
+			playsound(src, 'sound/effects/meteorimpact.ogg', 100, 1)
+			src.visible_message("<span class='warning'>[src] ploughs through [M], pulverizing it!</span>")
+			qdel(Q)
+
+/obj/mecha/combat/tank/Bumped(atom/movable/AM)
+	if(occupant)//can't run people over without a driver
+		if(ismob(AM))
```
Comment (Jc1469) [inline code/game/mecha/combat/tank.dm:120]: isliving :thinking:

**Comment by fonedsky** [comment]
I crown this tank... "The Holy Knight of Deadchat Spam"

**Comment by ghost** [comment]
>being such a shitty driver that you get in a car accident with ghosts

fund it

File: code/game/mecha/combat/tank.dm (hunk 1 above)
Comment (Time-Green) [reply code/game/mecha/combat/tank.dm]: So everytime it moves a 100 volume sound is played to literally every player?
🤔

**Pull request closed without merging**